### ISSUE TITLE
False positive: missing underscore error for URLs in download directive

### ISSUE THREAD

**Issue by VarunNikam22** [sphinx-contrib/sphinx-lint#134]
## Description

Sphinx-lint incorrectly reports a "_missing-underscore-after-hyperlink_" error for URLs used within inline `download` directives, even though these URLs are properly formatted and don't require trailing underscores according to reST syntax.

## Expected Behavior
Sphinx-lint should recognize that URLs within download directives are valid and not require trailing underscores, as they are not inline link references.

```rst
:download:`filename.txt <https://example.com/file.txt>`
```

**Comment by hugovk** [comment]
Thanks for the report.

The error is:

```
missing underscore after closing backtick in hyperlink (missing-underscore-after-hyperlink)
```

Docs ref showing this is valid: https://www.sphinx-doc.org/en/master/usage/referencing.html#role-download

The check is here:

https://github.com/sphinx-contrib/sphinx-lint/blob/9c97dd9b26f393464c3375f8cdc065c2c9ff3839/sphinxlint/checkers.py#L248-L259

Would you like to look into create a PR to fix?
